src/api/transactionSubmission.ts

```diff
@@ -110,6 +110,32 @@ export class TransactionSubmission {
     return signTransaction({ ...args });
   }
 
+  /**
+   * Sign a transaction that can later be submitted to chain, as a fee payer. The signer who signs the transaction
+   * will be the one who pays the gas
+   *
+   * @param args.signer The signer or fee payer
+   * @param args.transaction A raw transaction type (note that it holds the raw transaction as a bcs serialized data)
+   * ```
+   * {
+   *  rawTransaction: Uint8Array,
+   *  secondarySignerAddresses? : Array<AccountAddress>,
+   *  feePayerAddress?: AccountAddress
+   * }
+   * ```
+   *
+   * @return The signer AccountAuthenticator
+   */
+  /* eslint-disable class-methods-use-this */
+  signTransactionAsPayer(args: { signer: Account; transaction: AnyRawTransaction }): AccountAuthenticator {
```

Comment:
I feel this whole function can simply be a boolean arguments

```
signTransaction(args: { signer: Account; transaction: AnyRawTransaction, asSponsor?:true }): AccountAuthenticator {
if(asSponsor){
 transaction.feePayerAddress = signer.accountAddress;
}
    return signTransaction({ ...args });
  }
```

Reply:
Yeah I was thinking about this vs a new function. I wonder what everyone's preference are. We can sync up real quick on monday standup.